pkg/apis/componentconfig/types.go

```diff
@@ -111,6 +111,9 @@ type KubeSchedulerConfiguration struct {
 	// Indicate the "all topologies" set for empty topologyKey when it's used for PreferredDuringScheduling pod anti-affinity.
 	// DEPRECATED: This is no longer used.
 	FailureDomains string
+
+	// EnablePodFitsOnNodeOptimization enables shorting circuit all predicates if one predicate fails
+	EnablePodFitsOnNodeOptimization bool
```

Comment:
This is a bad name ... should more specific like `QuickFailPredicate` or something else

Reply:
I don't think we need an option for this. This should be default behavior in my opinion. It doesn't make sense to check all other predicates when one fails. We are adding the logic to order predicates and default ordering is to run predicates with lower overhead first. The whole ordering of predicates makes sense only if we stop checking predicates when one fails. Otherwise, the ordering is not useful.

Reply:
UPDATE: reading another thread, we may actually need a flag for those customers who want to find what are all the predicates that have failed for a node. So, let's rename this flag to something like `AlwaysCheckAllPredicates`. Default value should be false. Please add a commend in front of the flag that enabling this option may hurt scheduler performance.

Reply:
Totally agree this should put predicate ordering into consideration.

cc @wgliang It's https://github.com/kubernetes/community/pull/1152/files

Reply:
Sorry, it's my fault I should say that put predicate ordering will improve performance even more. In fact, we have already sorted out our own scheduler, for example, we will calculate the smaller number of CPU cores, memory size in front of the complex information such as disk and network. In this way, the implementation of the previous failure, to avoid later more complex calculations.

Reply:
Also, instead of modifying my current default behavior, I wanted to add a higher-performance option, albeit losing some of my failure information.

Reply:
@resouer You are right, I will rename it.

Reply:
+1 for default behaviour setting `AlwaysCheckAllPredicates` to false.